phpseclib/Math/BigInteger.php

```diff
@@ -2315,9 +2315,9 @@ function _montgomeryMultiply($x, $y, $m)
         $a = array(self::VALUE => $this->_array_repeat(0, $n + 1));
         for ($i = 0; $i < $n; ++$i) {
             $temp = $a[self::VALUE][0] + $x[$i] * $y[0];
-            $temp = $temp - self::$baseFull * (self::$base === 26 ? intval($temp / 0x4000000) : ($temp >> 31));
-            $temp = $temp * $cache[self::DATA][$key];
-            $temp = $temp - self::$baseFull * (self::$base === 26 ? intval($temp / 0x4000000) : ($temp >> 31));
+            $temp -= self::$baseFull * (self::$base === 26 ? (int) ($temp / 0x4000000) : ($temp >> 31));
```

Comment:
I checked, this PR can be merged into 1.0, so I'll just change target branch when review finished.